Intersect.Server/Entities/Combat/Dash.cs

```diff
@@ -50,40 +51,63 @@ public partial class Dash
             bool zdimensionPass = false
         )
         {
-            var n = 0;
             en.MoveTimer = 0;
             Range = 0;
             for (var i = 1; i <= range; i++)
             {
-                n = en.CanMove(Direction);
-                if (n == -5) //Check for out of bounds
+                if (!en.CanMoveInDirection(Direction, out var blockerType, out var entityType))
```

Comment:
Invert, don't add nesting

Reply:
But you said: https://github.com/AscensionGameDev/Intersect-Engine/pull/1775#discussion_r1157976094

and if we invert this -> https://github.com/AscensionGameDev/Intersect-Engine/pull/1775#discussion_r1157939900

also if we invert this, where do we place this within the for chain? inverting would mean to use the following early:

```csharp
                en.Move(Direction, null, true);
                en.Dir = Facing;
                Range = i;
```

this would make the switches... useless in the logic chain (unless we duplicate these 3 lines at the very end of the for method then but this doesn't sounds efficient...), from what i'm seeing this can be either done by using the negated version of the boolean (like it is atm) or either don't use the boolean return type which is what you pointed out to be changed in the previous review.

Reply:
Correct, it would duplicate the lines at the end. Disregard.